### PR TITLE
✨ Add survey icon to survey in trail page

### LINKED ISSUE #715: use image_url from survey object at trails page
The user can register an `icon url` when he's creating a new survey. This `icon_url` when it exists, it'll be display an icon at trails page. Now, survey object has the optional field `icon_url`
image

The `icon_url` is in survey object and surveys object is in trail:
image

The `icon_url` in trail can be found in:
image

### PR DESCRIPTION
**Pull request by iurynogueira**
fix #715 

# Add survey icon to survey in trail page

## Type of changes

Please delete options that are not relevant.

- [ ] **Bug fix**
- [x] **New feature**
- [ ] **Breaking change**

---

## Checklist:

- [x] My code is understandable
- [x] My code follows the styleguide of this project
- [x] My changes don't generate new warnings/errors
- [x] I have performed a self-review of my own code (build/lint/tests)

  ### Frontend

  - Components:

    - [x] I have implemented/updated unit tests
    - [ ] I have implemented/updated Storybook documentation

  - Pages:
    - [ ] I have implemented/updated E2E tests

### CODE REVIEW

File: frontend/src/components/roadmap-survey/roadmap-survey.styles.ts [hunk 1]
```diff
@@ -20,11 +20,14 @@ export const SurveyItemMap = styled.div<{ status: StatusGroup }>`
   }
 `;
 
-export const IconItem = styled.div<{ status: StatusGroup }>`
+export const IconItem = styled.div<{ status: StatusGroup, icon_url: string }>`
   height: 80px;
   width: 80px;
   border-radius: 50%;
-  background: ${(props) => getAvailableColor(props.status)};
+  background-image: url(${(props) => props.icon_url || '/assets/survey.png'});
```
Comment (edimossilva) [inline frontend/src/components/roadmap-survey/roadmap-survey.styles.ts:27]: Maybe rename this file to `default_survey_icon.png` or `survey_icon.png`

File: frontend/src/components/roadmap-survey/roadmap-survey.test.tsx [hunk 2]
```diff
@@ -29,22 +29,36 @@ describe('RoadmapSurvey', () => {
...
+      'border: 4px solid gray'
     );
   });
 
   it('should not render groups when not passed and show message', () => {
     render(<RoadmapSurvey groups={[]} />);
     expect(screen.getByText('No surveys to show =/')).toBeInTheDocument();
   });
+
+  it('should render survey icon background', () => {
```
Comment (edimossilva) [inline frontend/src/components/roadmap-survey/roadmap-survey.test.tsx:51]: maybe `it should render "icon from icon_url" when the survey has  "icon_url"`

File: frontend/src/components/roadmap-survey/roadmap-survey.test.tsx [hunk 3]
```diff
@@ -29,22 +29,36 @@ describe('RoadmapSurvey', () => {
...
+    render(<RoadmapSurvey groups={groups} />);
+    expect(screen.getByTestId(`icon-${groups[0].surveys[0].name}`)).toHaveStyle(
+      `background-image: url(${groups[0].surveys[0].icon_url})`
+    );
+  });
+
+  it('should render survey default icon background when dont have icon', () => {
```
Comment (edimossilva) [inline frontend/src/components/roadmap-survey/roadmap-survey.test.tsx:58]: maybe `it should render "default survey icon" when the survey has no "icon_url"`

**Review by edimossilva — APPROVED** [review]
LGTM!
I've made a few suggestions about some naming conventions, definitely not blockers but I believe they bring some small improvements

File: frontend/src/components/roadmap-survey/roadmap-survey.styles.ts [hunk 4]
```diff
@@ -20,11 +20,14 @@ export const SurveyItemMap = styled.div<{ status: StatusGroup }>`
   }
 `;
 
-export const IconItem = styled.div<{ status: StatusGroup }>`
+export const IconItem = styled.div<{ status: StatusGroup, icon_url: string }>`
```
Comment (andersonmalheiro) [inline frontend/src/components/roadmap-survey/roadmap-survey.styles.ts:23]: I suggest creating an interface for this

**Review by andersonmalheiro — CHANGES_REQUESTED** [review]

**Comment by iurynogueira** [comment]
Thank you guys!

**Review by mizek1 — APPROVED** [review]

**Pull request merged**